### ISSUE TITLE
Subscriber Equipment is lost to the user after a Reset

### ISSUE THREAD

**Issue by LB7979** [HabitRPG/habitica#3349]
When a user does a Rebirth, they should be able to have access to all Equipment they previously unlocked right?

But apparently when a subscribed user does a Rebirth, Subscriber Equipment they previously got is not available anymore.

I was a subscriber in March (and still am now) so I got the Forest Antlers.
Then I did a Reset, and today I got the Butterfly Subscriber Equipment of April, but March's Subscriber Equipment (Forest Antlers) is not there to me anymore.

All the Seasonal (Winter Wonder, Spring Fling etc.) Equipment is re-unlockable after a Rebirth right, so shouldn't the monthly Subscriber Equipment be too?

**Comment by LB7979** [comment]
Sorry by Subscriber Equipment I mean the Mystery Items.
Also, ID = `ed0f12b8-f273-4378-bb10-60a8a1638c16`

**Comment by deilann** [comment]
Hmm. When I did a rebirth my February Subscriber items wre repurchasable.

**Comment by LB7979** [comment]
Do you mean they were just in your Rewards column?

They're not in my 'Rewards' column on the Tasks page, and also not on the Equipment page. I see the April/Butterfly equipment on the Equipment page, but not the March/Antler equipment that I had before my Reset.

Also, since you say 're _purchas_ able' do you mean you had to spend Gold to get your subscriber items again?
That would be very strange. Since the point of the subscriber items is that you get them for donating real-life money to the developers, not for in-game Gold you spent on it. If you did not spend Gold to get the subscriber item the first time, why would it suddenly cost Gold after a Rebirth?

Also note: I was using Rebirth and Reset interchangably here, but in my case I did a Reset Account, not Rebirth. Don't know if that made the difference, but in both cases, shouldn't you be able to get back equipment you had before, for the same price you payed for it before?

**Comment by deilann** [comment]
I repurchased them for 0 GP.

...and yes, reset account is very different. Reset _resets_ your account.
It doesn't maintain your old user data, like having previously owned
Mystery Gear.

**Comment by LB7979** [comment]
Hm, OK. But if a Reset _doesn't maintain old user data_, why am I a Mage on Level 1 after my Reset? When a user starts to play HRPG initially, they start as a Warrior and don't get to choose a class until level 10. Yet when I did the Reset, I remainded a Mage at level 1 (which surprised me, I expected to be turned into a Warrior again until reaching level 10 again).
Also, I have a Mana bar now at level 1 after the Reset (even though I don't have the ability to _spend_ Mana yet). A first-time HRPG user does not have a Mana bar util later (until they unlock Class, I think).

What I'm saying is: a Reset does obviously not erase _all_ user data, because after a Reset you still keep Class, Mana bar and your Pets amongst other data. So it doesn't seem correct that it erases the fact that the user previously had Subscriber Items, that should be one of the things a Reset _retains_!

BTW, does this also mean that I can't re-purchase the Spring Fling items again?

**Comment by deilann** [comment]
There's a reason reset is in the "Danger Zone" and is a red button.

And yes, it means you will not be able to re-purchase Spring Fling items.
Reset gets rid of all of your tasks and task history, all of your equipment
(just wipes equipment, not "breaks" it like Rebirth), your gold, and XP. It
does _not_ change class partially because it's not a well-maintained
feature and partially because changing class costs gems, so it doesn't
touch that user data.

There is absolutely nothing I can do about this. Someone with database
access might be able to do something.

**Comment by LB7979** [comment]
Well it's not even that I'm specifically asking for to get the Forest Antlers back now, so that's okay. At this point, I just want to bring this under the attention of the developers (so the following is not directed at you personally Deilann, but a general thing for the consideration of all involved in HabitRPG's development):

The reason that some things, like Pets and Class, are UNaffected by a Reset is because they cost Gems, and getting Gems involves real-life money (either through subscription or a one-time donation).
Mystery items are something the user gets in exchange for giving real-life money too.
So it's still wrong that _some_ things that involve Gems and thus real-life money are retained, yet some others (mystery items) are not.

If the developers' philosophy is "items that the user might have previously spent Gems on should be unaffected, since they cost real-life money" then that should also apply to mystery items!

**Comment by deilann** [comment]
Lisa, that's the logic for rebirth -- which is a separate feature from
reset.

**Comment by deilann** [comment]
(I'm do want to point out that doesn't mean it _shouldn't_ be the logic for
reset. The reason a lot of this isn't part of reset is that reset isn't
well-maintained, so it was just never added into the code. There is no
logic at all for how these things work with reset.)

**Comment by lemoness** [comment]
That being said - I do think that we should get you back your subscriber items at the very least. Also, perhaps Reset should have a better-written warning?

**Comment by deilann** [comment]
@lemoness I don't know if reset needs a better written warning. The whole idea of an account reset is to well, reset it. Back to basics. It doesn't do what it's supposed to at the moment, which is to wipe everything. And it's in the "Danger Zone" right next to "Delete Account." In my opinion, it would be like putting a better warning on delete account.

But I do agree this user should get their items back. But @lefnire's gonna hafta do that.

**Comment by lemoness** [comment]
Maybe it should at least say that Rebirth is an option? Like, "Warning:
this will fully reset your account! If you would rather start over from the
beginning with tasks and rewards, but keep pets and gem-purchased items,
use Rebirth instead."

**Comment by lemoness** [comment]
(I know that right now Reset keeps pets, but that doesn't seem to be fully
intentional.)

**Comment by deilann** [comment]
Ahh, that would be good.

**Comment by Alys** [comment]
https://github.com/HabitRPG/habitrpg-shared/pull/245 incorporates feedback from this issue

**Comment by lefnire** [comment]
Oh shoot, confirmed! See [index.coffee#408](https://github.com/HabitRPG/habitrpg-shared/blob/9e85a886ba323b6bcbd1cbf91b80ca386b920d6e/script/index.coffee#L408). How does this look (devs, cc @SabreCat):

``` diff
diff --git a/script/index.coffee b/script/index.coffee
index 229d13d..b55e13d 100644
--- a/script/index.coffee
+++ b/script/index.coffee
@@ -405,7 +405,8 @@ api.wrap = (user, main=true) ->
           gear[type].weapon = 'weapon_base_0'
           gear[type].head   = 'head_base_0'
           gear[type].shield = 'shield_base_0'
-        user.items.gear.owned = {weapon_warrior_0:true}
+        _.each user.items.gear.owned.toObject(), (v,k)->user.items.gear.owned[k]=false
+        user.items.gear.owned.weapon_warrior_0 = true
         user.markModified? 'items.gear.owned'
         user.preferences.costume = false
         cb? null, user
```

Once I get this up, @LB7979 I'll restore your stuff

**Comment by Alys** [comment]
But isn't the point of a reset that it's meant to remove (almost)
everything? It starts an account from scratch again. If a player wants to
restart while being able to buy back all gear, then they can use the Orb of
Rebirth instead of reset.

**Comment by lefnire** [comment]
IMO, anything purchased with IRL $ (subscriber gear) or earned externally (contrib gear) shouldn't be touched by reset, plus it's not part of the earn system. The above diff will only send that gear to the reward store anyway, so they still have to buy it all back. 

Sounds like some varying opinions here, maybe we should have a vote on this?

**Comment by Alys** [comment]
I think the core problem with this issue was that reset wasn't documented
clearly enough to allow the player to realise exactly what would happen. If
anything, I think reset should become more powerful so that it removes pets
and class as well - it would become a true reset for those that really
wanted it.

The problem with losing items you've paid money for is a good point,
but if reset is documented clearly then it's the user's own choice.
Although in LB7979's case, I'm certainly not arguing against a fix for her
gear, since the documentation wasn't very detailed when she used it.

Also be cautious about making an Orb of Rebirth valueless. If reset has
very similar functions to Rebirth, few people will spend gems on Rebirth.

**Comment by lefnire** [comment]
Seems there a couple ways we can better Reset. Another is they can use it only after initial "tutorial" usage of Habit (I've seen this in other games), so it's a modal of "ready to begin?" and remove the button from settings altogether, favoring Rebirth from then on out. Another point is that none of what we're discussing here devalues Rebirth, which gives you an achievement and on-avatar icon (visible to all) plus some ingame bonuses.

However, the issue at hand is two possible (non-mutually-exclusive) quick-fixes to Reset. (1) better documentation, (2) don't remove purchased/contrib/etc gear (https://github.com/HabitRPG/habitrpg-shared/commit/9f6230b4d1de78a9bab9de54fab0c31f5dfb7945). @Alys I'm gathering you're not in favor of 2 as an additional quick-fix commit. I am, so mayhap let's create a poll for this one. (Anyone know how to make wikia polls? I never got the hang of it)

**Comment by Alys** [comment]
No, do go ahead with the quick-fix commit! You're the boss. :) I don't
think it's worth the effort of a poll. I don't really mind one way or the
other, I just thought reset was intended to destroy-almost-all-the-things.
:)

In a few hours when I get home, I'll submit another change to the
documentation to adjust the part about what happens to the gear.

**Comment by lefnire** [comment]
Aw... didn't mean to brow-beat ya, figured there would be disagreement so I wanted to argue my case and hear any contentions / votes (esp. Sabe who did Rebirth) - sorry about that! Ok, I'll push and fix @LB7979

**Comment by Alys** [comment]
You didn't brow-beat!  Your comments about the Rebirth achievement etc
convinced me - I'd forgotten about those. :) I should have mentioned that
in my previous comment.

**Comment by SabreCat** [comment]
It makes sense to me to continue with Reset wiping out everything but paid-for stuff, given clear documentation. If a user wants to really-truly start from scratch, they can achieve that with Delete Account, aye? (With some inconveniences due to getting a new UUID and API token, granted, but we are talking about a pretty extreme action here.)

Didn't @deilann have a feature in the pipeline that'd let you "set your pets free" so you could reacquire them all? I'd still like to see that one someday ^.^

**Comment by deilann** [comment]
I do!

I also think that reset should... reset your account, though. Because
unless reset does that, there's no way to do it if a player wants.

**Comment by Alys** [comment]
lefnire's code change from [his comment above](https://github.com/HabitRPG/habitrpg/issues/3349#issuecomment-48257764) has been made so I've updated the Reset documentation again:

https://github.com/HabitRPG/habitrpg-shared/pull/246
